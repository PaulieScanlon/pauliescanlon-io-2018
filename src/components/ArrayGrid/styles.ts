import styled from "react-emotion";

import { common } from "../../styles/";

export const ArrayGridWrapper = styled.div({
  label: "array-grid-wrapper",
  display: "flex",
  flexWrap: "wrap",
  margin: `0 -${common.padding / 2}px 0 -${common.padding / 2}px`
});
