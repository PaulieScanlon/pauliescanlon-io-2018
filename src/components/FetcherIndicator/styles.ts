import styled from "react-emotion";

export const IndicatorWrapper = styled.div({
  label: "indicator-wrapper",
  with: "100%",
  minHeight: 300
});
