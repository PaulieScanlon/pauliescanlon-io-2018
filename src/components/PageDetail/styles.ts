import styled from "react-emotion";

export const PageDetailWrapper = styled.div({
  label: "page-detail-wrapper"
});
