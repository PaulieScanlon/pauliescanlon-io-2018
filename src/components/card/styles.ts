import styled from 'react-emotion';

export const CardWrapper = styled.div({
  label: 'card-wrapper'
});
