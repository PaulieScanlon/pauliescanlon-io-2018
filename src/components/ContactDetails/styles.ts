import styled from "react-emotion";
import { draculaTheme } from "../../styles/theme";
import { common } from "../../styles/";

export const ContactDetailsWrapper = styled.div({
  label: "contact-details-wrapper",
  display: "flex",
  flexDirection: "column"
});

export const ContactLink = styled.a({
  label: "contact-link",
  cursor: "pointer",
  fontSize: 14,
  color: draculaTheme.comment,
  transition: `all ${common.hoverTransition}s`,
  "&:hover": {
    color: draculaTheme.purple
  }
});
