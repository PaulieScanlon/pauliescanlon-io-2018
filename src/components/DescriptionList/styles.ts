import styled from "react-emotion";
import { draculaTheme } from "../../styles/theme";

export const DL = styled.dl({
  label: "descriptions-list"
});

export const DT = styled.dt({
  label: "descriptions-term",
  color: draculaTheme.cyan
});

export const DD = styled.dd({
  label: "descriptions-description",
  margin: "0px 0px 10px 0px",
  color: draculaTheme.comment,
  a: {
    color: `${draculaTheme.comment}!important`
  }
});
