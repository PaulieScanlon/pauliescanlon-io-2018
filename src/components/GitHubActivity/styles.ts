import styled from "react-emotion";
import { draculaTheme } from "../../styles/theme";
import { common } from "../../styles/";

export const GitHubActivityWrapper = styled.div({
  label: "git-hub-activity-wrapper",
  display: "flex",
  flexDirection: "column"
});

export const RepoLink = styled.a({
  label: "repo-link",
  cursor: "pointer",
  fontSize: 14,
  color: draculaTheme.comment,
  transition: `all ${common.hoverTransition}s`,
  "&:hover": {
    color: draculaTheme.purple
  }
});
