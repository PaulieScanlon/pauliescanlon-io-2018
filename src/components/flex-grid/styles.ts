import styled from 'react-emotion';

// import { IPx } from '../../styles/breakpoints';
import { mq } from '../../styles/breakpoints';

interface IStyles {
  xs?: number;
  sm?: number;
}

export const FlexContainer = styled.div({
  label: 'flex-container',
  display: 'flex',
  flexWrap: 'wrap'
});

export const FlexItem = styled.div<IStyles>(
  {
    label: 'flex-item',
    position: 'relative',
    boxSizing: 'border-box'
  },
  ({ xs, sm }) => ({
    [mq[0]]: {
      border: '1px solid red',
      width: `calc(${100 / xs}% - ${0}px)`
    },
    [mq[1]]: {
      border: '1px solid blue',
      width: `calc(${100 / sm}% - ${0}px)`
    }
  })
);
