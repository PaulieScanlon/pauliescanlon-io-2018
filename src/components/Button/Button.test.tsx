import * as React from "react";
import { shallow, mount } from "enzyme";

import { Button } from "./";
import { draculaTheme } from "../../styles/theme";

describe("<Button />", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Button>Lorem Ipsum</Button>);
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders children", () => {
    const wrapper = mount(<Button>Lorem Ipsum</Button>);
    expect(wrapper.props().children).toEqual("Lorem Ipsum");
  });
  it("has correct background colour", () => {
    const wrapper = mount(
      <Button backgroundColour={draculaTheme.green}>Lorem Ipsum</Button>
    );
    expect(wrapper.props().backgroundColour).toEqual(draculaTheme.green);
  });
});
