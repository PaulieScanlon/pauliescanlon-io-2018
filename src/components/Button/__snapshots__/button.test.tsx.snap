// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button>
    Hooray
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Hooray",
      "className": "css-1rsanyl-Button-button e93mxpc0",
    },
    "ref": null,
    "rendered": "Hooray",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hooray",
        "className": "css-1rsanyl-Button-button e93mxpc0",
      },
      "ref": null,
      "rendered": "Hooray",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
